d2common/d2math/d2vector/vector_test.go

```diff
@@ -7,50 +7,17 @@ import (
 	"github.com/OpenDiablo2/OpenDiablo2/d2common/d2math"
 )
 
-/*func TestMain(m *testing.M) {
-	setup()
-	os.Exit(m.Run())
-}*/
-
-var outVector Vector
-var outFloat float64
-
-/*func setup() {
-}*/
-
-// TODO: Remove these evaluate functions. Throwing test errors outside the relevant functions means otherwise handly links to failed tests now point here which is annoying.
-
-func evaluateVector(description string, want, got Vector, t *testing.T) {
-	if !got.Equals(want) {
-		t.Errorf("%s: wanted %s: got %s", description, want, got)
-	}
-}
-
-func evaluateVectorApprox(description string, want, got Vector, t *testing.T) {
-	if !got.EqualsApprox(want) {
-		t.Errorf("%s: wanted %s: got %s", description, want, got)
-	}
-}
-
-func evaluateScalar(description string, want, got float64, t *testing.T) {
-	if want != got {
-		t.Errorf("%s: wanted %f: got %f", description, want, got)
-	}
-}
-
-func evaluateScalarApprox(description string, want, got float64, t *testing.T) {
-	if d2math.CompareFloat64Fuzzy(want, got) != 0 {
-		t.Errorf("%s: wanted %f: got %f", description, want, got)
-	}
-}
-
-func evaluateChanged(description string, original, clone Vector, t *testing.T) {
-	if !original.Equals(clone) {
-		t.Errorf("%s: changed vector %s to %s unexpectedly", description, clone, original)
-	}
-}
+//nolint:gochecknoglobals // These variables are assigned to in benchmark functions to avoid compiler optimisations
```

Comment:
I initially ran into what I assume is the compiler optimisation issue and assigning to a package level variable does resolve it. It doesn't seem to be necessary when operating on the vector itself, only when returning a value which is not used. Assigning to _ does not work.

I don't think it's possible to avoid the unused variable warnings without either adding an unused function which assigns them to something (no) or some sort of over the top hackery. 

Happy to hear about any better alternatives to all this. Since it's in a _test.go file I guess it isn't really the end of the world.